### ISSUE TITLE
ec2_lc fail to create when no default vpc in region

### ISSUE THREAD

**Issue by mogadanez** [ansible/ansible-modules-core#1719]
##### Issue Type:

 “Bug Report”
##### Ansible Version:

1.9.2
##### Ansible Configuration:

basic configuration
##### Environment:

N/A
##### Summary:

ec2_lc fail to create when no default vpc in region
##### Steps To Reproduce:
1.  delete default vpc in some region
2. create non-default vpc by hand or ec2_vpc module
3. try run 

```
- ec2_lc:
    name: launch config
    state: present
    image_id: any
    region: "{{ region }}"
    key_name: any
    security_groups: ['any group doesnt matter' ]
    instance_type: t1.micro
```
##### Expected Results:

Launch configuration  should be created .
like:
awscli create launch config without  any errors.
aws web console show same error but anyway create config.
##### Actual Results:

error happen

```
failed: [127.0.0.1] => {"failed": true}
msg: BotoServerError: 400 Bad Request
<ErrorResponse xmlns="http://autoscaling.amazonaws.com/doc/2011-01-01/">
  <Error>
    <Type>Sender</Type>
    <Code>ValidationError</Code>
    <Message>No default VPC for this user</Message>
  </Error>
  <RequestId>a95979d7-28d0-11e5-a457-7b851bc22db7</RequestId>
</ErrorResponse>
```

**Comment by bcoca** [comment]
Can You Help Us Out?
====================

Thanks for filing a ticket! I am the friendly GitHub Ansibot.

It looks like you might not have filled out the issue description based on our standard issue template.  You might not have known about that, and that's ok too, we'll tell you how to do it.

We have a standard template because Ansible is a really busy project and it helps to have some standard information in each ticket, and GitHub doesn't yet provide a standard facility to do this like some other bug trackers.  We hope you understand as this is really valuable to us!.

Solving this is simple: please copy the contents of this [template](https://raw.githubusercontent.com/ansible/ansible/devel/ISSUE_TEMPLATE.md) and **paste it into the description** of your ticket.  That's it!

If You Had A Question To Ask Instead
====================================

If you happened to have a "how do I do this in Ansible" type of question, that's probably more of a user-list question than a bug report, and you should probably ask this question on the project mailing list instead.

However, if you think you have a bug, the report is the way to go!  We definitely want all the bugs filed :)  Just trying to help!

About Priority Tags
===================

Since you're here, we'll also share some useful information at this time.

In general tickets will be assigned a priority between P1 (highest) and P5, and then worked in priority order. We may also have some follow up questions along the way, so keeping up with follow up comments via GitHub notifications is a good idea.

Due to large interest in Ansible, humans may not comment on your ticket immediately.

Mailing Lists
=============

If you have concerns or questions, you're welcome to stop by the ansible-project or ansible-development mailing lists, as appropriate.  Here are the links:

  * https://groups.google.com/forum/#!forum/ansible-project - for discussion of bugs and how-to type questions
  * https://groups.google.com/forum/#!forum/ansible-devel - for discussion on how to implement a code change, or feature brainstorming among developers

Thanks again for the interest in Ansible!

**Comment by mhbrown** [comment]
I'm having this same issue with ansible 1.9.3. @bcoca do you know if this is a bug, or is there some variable I should add?

**Comment by mhbrown** [comment]
Duplicate of https://github.com/ansible/ansible-modules-core/issues/2035?

**Comment by raben2** [comment]
The issue only appears if you use the name of the security_group when using id everything works fine.

**Comment by mhbrown** [comment]
Thanks @raben2! Looks like it's working when I use a list of security group ids rather than names.

I suggest the documentation be updated at http://docs.ansible.com/ansible/ec2_lc_module.html to make it more obvious that the list of security groups should be ids, and not names.

**Comment by raben2** [comment]
I agree but using names on the sg´s makes the deployment much more flexible. Since you don´t have to dig into the output json of other tasks. ;)

**Comment by mhbrown** [comment]
Agreed

**Comment by filipenf** [comment]
@mogadanez I've created PR #4061 which fix this and also allows to mix security-groups IDs/names when creating a launch config. However, I don't know if this can be backported to 1.9.x because it depends on `get_ec2_security_group_ids_from_names` which was added just recently.

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/30085
